lib/src/index_impl.dart

```diff
@@ -377,11 +382,11 @@ class MeiliSearchIndexImpl implements MeiliSearchIndex {
 
   @override
   Future<Map<String, List<String>>> getSynonyms() async {
-    final response = await http.getMethod('/indexes/$uid/settings/synonyms');
+    final response = await http
+        .getMethod<Map<String, Object?>>('/indexes/$uid/settings/synonyms');
```

Comment:
I did try that and it failed unfortunately, that's because the json decoder can't know the map type before creating it, so it just defaults to  `Map<String, Object?>`

Reply:
oh my bad, what I tried was `.getMethod<Map<String, List<String>>>` which should fail, but `.getMethod<Map<String, List>>` is fine